src/app/course/components/editor/editor.component.ts

```diff
@@ -0,0 +1,61 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {CourseFormControl, TCourse} from "../../models/course";
+import {CoursesService} from "../../services/CoursesService";
+
+export enum CourseEditorMode {
+	ADD = 'ADD',
+	EDIT = 'EDIT',
+}
+
+@Component({
+	selector: 'course-editor',
+	templateUrl: './editor.component.html',
+	styleUrls: ['./editor.component.scss']
+})
+export class CourseEditorComponent implements OnInit {
+	@Input() public mode: CourseEditorMode;
+	@Input() public courseId?: number;
+	public titleSubmit: string = 'Сохранить';
+	public titleEditor: string = 'Создание';
+	public course: CourseFormControl = new CourseFormControl();
+
+	private _courseService: CoursesService;
+
+	constructor(
+		courseService: CoursesService
```

Comment:
Try to use auto creation for properties by
```
constructor(private courseService: CoursesService) {}
```
that's all. Now you can call `this.courseService.***`